mbf_costmap_nav/src/mbf_costmap_nav/free_pose_search.cpp

```diff
@@ -283,30 +304,28 @@ SearchSolution FreePoseSearch::search() const
       // if the state is outside or unknown, we save the first one we find
       if ((tested_sol.search_state.state == SearchState::OUTSIDE ||
            tested_sol.search_state.state == SearchState::UNKNOWN) &&
-          !outside_or_unknown)
+          !no_info_sol)
       {
         ROS_DEBUG_STREAM_NAMED(LOGNAME.data(), "Found unknown/outside pose: " << tested_sol.pose.x << ", "
                                                                               << tested_sol.pose.y << ", "
                                                                               << tested_sol.pose.theta);
-        sol = tested_sol;
-        outside_or_unknown = true;
+        no_info_sol = tested_sol;
       }
 
       ROS_DEBUG_STREAM_COND_NAMED(tested_sol.search_state.state == SearchState::LETHAL, LOGNAME.data(),
-                                  "Footprint in cell " << cell.x << ", " << cell.y << "; pose: (" << sol.pose.x << ", "
-                                                       << sol.pose.y << ", " << sol.pose.theta
+                                  "Footprint in cell " << test_cell.x << ", " << test_cell.y << "; pose: ("
+                                                       << sol.pose.x << ", " << sol.pose.y << ", " << sol.pose.theta
                                                        << ") is in lethal obstacle or unknown; skipping");
     }
-
     else
     {
-      ROS_DEBUG_STREAM_NAMED(LOGNAME.data(), "Cell: (" << cell.x << ", " << cell.y << ") with pose: (" << sol.pose.x
-                                                       << ", " << sol.pose.y << ", " << sol.pose.theta
+      ROS_DEBUG_STREAM_NAMED(LOGNAME.data(), "Cell: (" << test_cell.x << ", " << test_cell.y << ") with pose: ("
+                                                       << sol.pose.x << ", " << sol.pose.y << ", " << sol.pose.theta
                                                        << ") is in lethal obstacle or in unknown space; skipping");
     }
 
     // adding neighbors to queue
-    const std::vector<Cell> neighbors = getNeighbors(*costmap2d, cell);
+    const std::vector<Cell> neighbors = getNeighbors(*costmap2d, test_cell);
     for (const Cell& neighbor : neighbors)
```

Comment:
```suggestion
    for (const auto& neighbor : neighbors)
```
S1: `auto`

---

S2: replace the following if statement

https://github.com/magazino/move_base_flex/blob/c44b78e17cf0a4d4063907a6f50beea31fb61e76/mbf_costmap_nav/src/mbf_costmap_nav/free_pose_search.cpp#L332

with

```cpp
if (in_queue_or_visited.count(cell_index))
```

This line is outside the PR so I can't comment on it :sweat_smile:

Reply:
In this case, they have equal performance, and the result is the same since the container is `std::unordered_set<int>`
But I prefer to use the algorithm with better intent (better expresses what I am doing).
If it was c++20, I would use `contains`

Reply:
I'd argue count also expresses clear intent: checking if an element exists in the container is the same as checking if there is at least 1 elem of it (i.e. count > 0). So since count is shorter / easier to read usually I prefer it over find. But it's just a suggestion so feel free to ignore :shrug: